### DISCUSSION TITLE
How extend comments with custom highlights in nvim >=0.8.0?

### DISCUSSION THREAD

**Discussion by erikw** [nvim-treesitter/nvim-treesitter#3729] [General]
Hey,

before upgrading nvim and nvim-treesitter to 0.8.0, or maybe some change that happended recently after this in nvim-treesitter, I could add a custom highlight in comments by creating a file like this

`$XDG_CONFIG_DIR/nvim/queries/comment/highlights.scm`
```scm
("text" @text.danger
 (#any-of? @text.danger "NOPE" ))
```

Currently this will add the new `NOPE` comment marker, but the built-in like TODO etc stops working that are defined in this plugin at 
[nvim-treesitter/queries/comment/highlights.scm](https://github.com/nvim-treesitter/nvim-treesitter/blob/master/queries/comment/highlights.scm)
meaning that the `NOPE` comment marker is the only one working.

It seems like now my custom file overwrites the default one. The [README](https://github.com/nvim-treesitter/nvim-treesitter/blob/master/README.md#adding-queries) says one should by convention use `after/queries` to extend. I tried this and then my custom file is not picked up at all.

I can't pin-point exactly when this broke but I would say in the last ~2-4 weeks.

To put this issue is a positive format instead; would there be an example of how one can correctly extend the base comment highlights with a custom one? :)

My system:
* Nvim: `0.8.0`
* OS: macOS
* nvim-treesitter: latest commit at master & `:TSUpdate`'d

**Comment by clason** [comment]
You need to mark the query as `;; extends`, see `:h treesitter-query-modeline-extends`.

PSA: Treesitter support is still experimental so breaking changes need to be expected. Make sure to follow the updates in https://github.com/neovim/neovim/issues/14090 and https://github.com/nvim-treesitter/nvim-treesitter/issues/2293.

**Comment by erikw** [comment]
Thanks @clason, that did it :)

For reference if someone else would wonder, this worked for me:


Contents of `$XDG_CONFIG_DIR/nvim/after/queries/comment/highlights.scm`
```scm
;; extends

("text" @text.danger
 (#any-of? @text.danger "NOPE" ))
```
